internal/lsp/watcher/rlimit_unix.go

```diff
@@ -0,0 +1,57 @@
+//go:build unix
+
+// This file contains code inspired by Syncthing's rlimit implementation
+// Syncthing is licensed under the Mozilla Public License Version 2.0
+// See: https://github.com/syncthing/syncthing/blob/main/LICENSE
+
+package watcher
+
+import (
+	"runtime"
+	"syscall"
+)
+
+const (
+	// macOS has a specific limit for RLIMIT_NOFILE
+	darwinOpenMax = 10240
+)
+
+// MaximizeOpenFileLimit tries to set the resource limit RLIMIT_NOFILE (number
+// of open file descriptors) to the max (hard limit), if the current (soft
+// limit) is below the max. Returns the new (though possibly unchanged) limit,
+// or an error if it could not be changed.
+func MaximizeOpenFileLimit() (int, error) {
```

Comment:
Same here. This function should not be public, unless we move to `fsext` or it's own standalone package.